src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationRecordPattern.java

```diff
@@ -0,0 +1,74 @@
+//non-compiled with javac: Compilable with Java21        //indent:0 exp:0
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;//indent:0 exp:0
+//indent:0 exp:0
+//indent:0 exp:0
+/* Config:                                                                  //indent:0 exp:0
+ * basicOffset = 4                                                          //indent:1 exp:1
+ * braceAdjustment = 0                                                      //indent:1 exp:1
+ * caseIndent = 4                                                           //indent:1 exp:1
+ * tabWidth = 4                                                             //indent:1 exp:1
+ * throwsIndent = 8                                                         //indent:1 exp:1
+ * forceStrictCondition = true                                             //indent:1 exp:1
+ */                                                                         //indent:1 exp:1
+public class InputIndentationRecordPattern {                                //indent:0 exp:0
+    record ColoredPoint(boolean p, int x, int c) { }       //indent:4 exp:4
+    record Rectangle(ColoredPoint upperLeft, ColoredPoint lowerRight) { }        //indent:4 exp:4
+
+    void test(Object obj) { //indent:4 exp:4
+        boolean b = obj instanceof //indent:8 exp:8
+                ColoredPoint(_,_,_);  //indent:16 exp:12 warn
+        boolean c = obj instanceof //indent:8 exp:8
+            ColoredPoint(_,_,_);  //indent:12 exp:12
+
+        boolean bb = obj instanceof //indent:8 exp:8
+        ColoredPoint(_,_,_);  //indent:8 exp:12 warn
+
+        if (obj instanceof ColoredPoint(_,_,_)) {} //indent:8 exp:8
+
+        if (obj instanceof //indent:8 exp:8
+                ColoredPoint(_,_,_)) {} //indent:16 exp:12 warn
+        if (obj instanceof     //indent:8 exp:8
+                ColoredPoint(_,_,_)) {} //indent:16 exp:12 warn
+
+        if (obj instanceof  //indent:8 exp:8
+        ColoredPoint(_,_,_)) {  } //indent:8 exp:12 warn
+
+        if (obj instanceof //indent:8 exp:8
+ColoredPoint(_,_,_)) { }  //indent:0 exp:12 warn
+        if (obj instanceof //indent:8 exp:8
+ColoredPoint( //indent:0 exp:12 warn
+_,  //indent:0 exp:12 warn
+_,  //indent:0 exp:12 warn
+_)) {} //indent:0 exp:12 warn
```

Comment:
I see a lot of examples with warns, but I don't see many with proper indentation. Pick a few, including this one, and show them with proper indentation right after the wrong ones.

Reply:
I typically make two new tests with the same input, one with no violations and one with violations

Reply:
done each example should have the correct one now in the same  file